### PR TITLE
Code: Implement inter-process synchronization for LiteDB

### LINKED ISSUE #10248: Code Quality : Duplication of FileTagHelper
There is currently two FileTagsHelper coexisting in the code. One in `Files.App.Helpers` and the other in `Files.App.Filesystem`. 

They share multiple methods with the same signatures, however their implementations are somewhat different.

The two helpers should be merged into only one contained in `Files.App.Helpers`.

Extra attention must be taken to ensure the merge of the shared methods.

### PR DESCRIPTION
**Pull request by hez2010**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #10248
- Fixes https://github.com/files-community/Files/pull/10008#pullrequestreview-1144304212

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/App.xaml.cs [hunk 1]
```diff
@@ -32,6 +32,7 @@
 using System;
 using System.Diagnostics;
 using System.Linq;
+using System.Threading;
```
Comment (QuaintMako) [inline src/Files.App/App.xaml.cs:35]: Only modification in the file it seems. Is it still relevant?

File: src/Files.App/ViewModels/FolderSettingsViewModel.cs [hunk 2]
```diff
@@ -380,12 +382,13 @@ private static void WriteLayoutPreferencesToDb(string folderPath, ulong? frn, La
...
-			if (DbInstance.GetPreferences(folderPath, frn) is null)
+            using var dbInstance = GetDbInstance();
+            if (dbInstance.GetPreferences(folderPath, frn) is null)
 			{
 				if (LayoutPreferences.DefaultLayoutPreferences.Equals(prefs))
 					return; // Do not create setting if it's default
 			}
-			DbInstance.SetPreferences(folderPath, frn, prefs);
+			dbInstance.SetPreferences(folderPath, frn, prefs);
 		}
```
Comment (QuaintMako) [inline src/Files.App/ViewModels/FolderSettingsViewModel.cs:392]: If formatting is a broken.

Comment (QuaintMako) [inline src/Files.App/ViewModels/FolderSettingsViewModel.cs:392]: If formatting is broken.

File: src/Files.App/ViewModels/ItemViewModel.cs [hunk 3]
```diff
@@ -1067,7 +1067,7 @@ public async Task LoadExtendedItemProperties(ListedItem item, uint thumbnailSize
 					ImageSource groupImage = null;
 					bool loadGroupHeaderInfo = false;
 					GroupedCollection<ListedItem> gp = null;
-					try
+                    try
```
Comment (QuaintMako) [inline src/Files.App/ViewModels/ItemViewModel.cs:1070]: The try indentation is not right.

File: src/Files.App/ViewModels/ItemViewModel.cs [hunk 4]
```diff
@@ -1101,7 +1101,10 @@ public async Task LoadExtendedItemProperties(ListedItem item, uint thumbnailSize
 										item.FileFRN = fileFRN;
 										item.FileTags = fileTag;
 									}, Microsoft.UI.Dispatching.DispatcherQueuePriority.Low);
-									FileTagsHelper.DbInstance.SetTags(item.ItemPath, item.FileFRN, item.FileTags);
+									using (var dbInstance = FileTagsHelper.GetDbInstance())
+									{
+										dbInstance.SetTags(item.ItemPath, item.FileFRN, item.FileTags);
+									}
```
Comment (QuaintMako) [inline src/Files.App/ViewModels/ItemViewModel.cs:1107]: This bit of code is used three times at least. Worth to be wrapped into a method `private void SetTagsOnDBInstance(ListedItem item)` ?

**Review by QuaintMako — COMMENTED** [review]
Don't have much experience with threads, but LGTM code-wise aside from the small formatting issues commented above.

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Review by QuaintMako — APPROVED** [review]
LGTM.

**Pull request merged**